src/main/java/com/github/junrar/rarfile/BaseBlock.java

```diff
@@ -159,6 +159,10 @@ public short getHeaderSize() {
         return headerSize;
     }
 
+    public short getHeaderPaddingSize() {
+        return (short) ((~headerSize + 1) & 0xF);
```

Comment:
it 's only used if the file is encrypted

Reply:
i mean, if i call this on a non-encrypted archive, what does it return ? I see in the calling code you check for encryption beforehand, but if this method would always return the appropruate value, you wouldn't need to check for isEncrypted in the calling code.

It should return 0 for unencrypted, and somethign for encrypted, no ?

Reply:
i konw waht you mean.
if it is called on  a non-encrypted archive,it will get wrong answer now.
we can add a field in BaseBlock to show if it is encrypted . then we can call it without  check for encryption.
how do you think?

Reply:
what does `(short) ((~headerSize + 1) & 0xF)` do exactly?

Reply:
If the header is encrypted, the header size is a multiple of 16.
so **(short) ((~headerSize + 1) & 0xF)** means padding.
for example:
if header size is 15 before encrypted,the size will be 16 after Encryption. ((~headerSize + 1) & 0xF) = 1.

When some headers are parsed, the data in the headers is not completely read.it  May only read 7 bytes, because the code does not care about other fields.
if so,when calculating the position of next block,，the expression is header startPos +  header size + padding size.

Reply:
does `getHeaderPaddingSize` need to be public after your refactoring?

Reply:
i also saw that the function `getHeaderSize()` is still used. Shouldn't we have a single public method? And don't you need to call the one with padding in some places where this one is called?

Reply:
> does `getHeaderPaddingSize` need to be public after your refactoring?

it's true.i will change it to be private.

Reply:
> i also saw that the function `getHeaderSize()` is still used. Shouldn't we have a single public method? And don't you need to call the one with padding in some places where this one is called?

i don't think so.user may be interested in header size. but they are not familiar with the new api.
We can unify the api in the project，but i think `getHeaderSize()` needs to be kept。
how do you think?

Reply:
This is indeed a public api but i don't think many people would use it.

In your code we still use it in some places, is it on purpose or you forgot something?